app/src/renderer/js/main.js

```diff
@@ -406,25 +303,48 @@ document.addEventListener('DOMContentLoaded', () => {
     app.kap.settings.set('dimensions', dimensions);
   };
 
-  const handleSizeChange = function () {
-    clearApp();
-    dimensions.ratio = [parseInt(this.value.split(':')[0], 10), parseInt(this.value.split(':')[1], 10)];
+  const handleSizeChange = function (ratio) {
+    dimensions.app = null;
+    dimensions.ratio = [parseInt(ratio.split(':')[0], 10), parseInt(ratio.split(':')[1], 10)];
 
     dimensions.ratioLocked = true;
     linkBtn.classList.add('is-active');
 
     if (dimensions.ratioLocked) {
-      dimensions.height = (dimensions.ratio[1] / dimensions.ratio[0]) * dimensions.width;
+      dimensions.height = Math.round((dimensions.ratio[1] / dimensions.ratio[0]) * dimensions.width);
       inputHeight.value = Math.round(dimensions.height);
     }
+    setCropperWindowSize(dimensions.width, dimensions.height);
+    dimensionsEmitter.emit('change', dimensions);
     app.kap.settings.set('dimensions', dimensions);
   };
 
-  aspectRatioSelector.addEventListener('change', handleSizeChange);
+  ipcRenderer.on('change-aspect-ratio', (e, aspectRatio) => handleSizeChange(aspectRatio));
+
+  dimensionsEmitter.on('ratio-selected', ratio => handleSizeChange(ratio));
+
+  dimensionsEmitter.on('app-selected', app => {
+    // Set app information on dimensions so it can be reused later
+    // eg. for showing app name after selection
+    dimensions.app = {
+      pid: app.pid,
+      fullscreen: app.fullscreen,
```

Comment:
Super nitpick, but `fullscreen` => `isFullscreen` (To make it clear when reading the code that it's a boolean).